modules/gapi/src/backends/fluid/gfluidcore_simd_sse41.hpp

```diff
@@ -554,7 +554,162 @@ CV_ALWAYS_INLINE void calcRowLinear_8UC_Impl_<3>(uint8_t* dst[],
+    bool yRatioEq1 = inSz.height == outSz.height;
+
+    constexpr int nlanes = v_float32x4::nlanes;
+
+    if (!xRatioEq1 && !yRatioEq1)
+    {
+        for (int line = 0; line < lpi; ++line) {
+            float beta0 = beta[line];
+            float beta1 = 1 - beta0;
+            v_float32x4 v_beta0 = v_setall_f32(beta0);
+            int x = 0;
+
+            v_float32x4 low1, high1, s00, s01;
+            v_float32x4 low2, high2, s10, s11;
+            for (; x <= outSz.width - nlanes; x += nlanes)
+            {
+                v_float32x4 alpha0 = v_load(&alpha[x]);
+                //  v_float32 alpha1 = 1.f - alpha0;
+
+                v_gather_pairs(src0[line], &mapsx[x], low1, high1);
+                v_deinterleave(low1, high1, s00, s01);
+
+                //  v_float32 res0 = s00*alpha0 + s01*alpha1;
+                v_float32x4 res0 = v_fma(s00 - s01, alpha0, s01);
+
+                v_gather_pairs(src1[line], &mapsx[x], low2, high2);
+                v_deinterleave(low2, high2, s10, s11);
+
+                //  v_float32 res1 = s10*alpha0 + s11*alpha1;
+                v_float32x4 res1 = v_fma(s10 - s11, alpha0, s11);
+
+                //  v_float32 d = res0*beta0 + res1*beta1;
+                v_float32x4 d = v_fma(res0 - res1, v_beta0, res1);
+
+                vx_store(&dst[line][x], d);
+            }
+
+            for (; x < outSz.width; ++x)
+            {
+                float alpha0 = alpha[x];
+                float alpha1 = 1 - alpha0;
+                int   sx0 = mapsx[x];
+                int   sx1 = sx0 + 1;
+                float res0 = src0[line][sx0] * alpha0 + src0[line][sx1] * alpha1;
+                float res1 = src1[line][sx0] * alpha0 + src1[line][sx1] * alpha1;
+                dst[line][x] = beta0 * res0 + beta1 * res1;
+            }
+        }
+
+    }
+    else if (!xRatioEq1)
+    {
+        GAPI_DbgAssert(yRatioEq1);
+
+        for (int line = 0; line < lpi; ++line) {
+            int x = 0;
+
+            v_float32x4 low, high, s00, s01;
+            for (; x <= outSz.width - nlanes; x += nlanes) {
+                v_float32x4 alpha0 = v_load(&alpha[x]);
+                //  v_float32 alpha1 = 1.f - alpha0;
+
+                v_gather_pairs(src0[line], &mapsx[x], low, high);
+                v_deinterleave(low, high, s00, s01);
+
+                //  v_float32 d = s00*alpha0 + s01*alpha1;
+                v_float32x4 d = v_fma(s00 - s01, alpha0, s01);
+
+                vx_store(&dst[line][x], d);
+            }
+
+            for (; x < outSz.width; ++x) {
+                float alpha0 = alpha[x];
+                float alpha1 = 1 - alpha0;
+                int   sx0 = mapsx[x];
+                int   sx1 = sx0 + 1;
+                dst[line][x] = src0[line][sx0] * alpha0 + src0[line][sx1] * alpha1;
+            }
+        }
+
+    }
+    else if (!yRatioEq1)
+    {
+        GAPI_DbgAssert(xRatioEq1);
+        int length = inSz.width;  // == outSz.width
```

Comment:
Is it necessary to keep `length` variable? If so, might it be worth using it in previous cases as well?